### DISCUSSION TITLE
Flux `Kustomization` - Mozilla SOPS and `secretRef` to another namespace

### DISCUSSION THREAD

**Discussion by mbrennan** [fluxcd/flux2#4290] [Q&A] [answered]
HI there, I'd like to have CertManager running in its own namespace (`cert-manager`).  I have a `ClusterIssuer` resource which references a CA secret, encrypted with Mozilla SOPS.  This secret also lives in the `cert-manager` namespace.  On the Flux `Kustomization` resource I have for CertManager, I reference the secret with `secretRef`.  However, this secret has to be in the same namespace as the `Kustomization` resource (in this case `cert-manager`) in order for the controller to decrypt the secret.  As far as I can tell, there is no way to reference another namespace from `secretRef`.  

[The documentation mentions](https://fluxcd.io/flux/components/kustomize/kustomizations/#controller-global-decryption) a way to uses "global decryption" if the secret can't be found in the namespace.  However, I can't seem to find a way to configure a secret say in the `flux-namespace` to be used globally - that would be ideal.

I'd like to keep my secret in `flux-namespace`, otherwise when I rebuild my cluster in an automated way the process breaks until I wait for the `cert-manager` namespace to exist, then deploy the secret to the `cert-manager` namespace.

Is there any way to reference a secret in `flux-system` with the `Kustomization` being in another namespace?

**Comment by makkes** [comment, marked as answer]
Global decryption only works with the providers mentioned in the doc you linked, i.e. AWS KMS, Azure Key Vault, GCP KMS or Hashicorp Vault.

**Comment by mloskot** [comment]
> Is there any way to reference a secret in flux-system with the Kustomization being in another namespace?

👉 https://github.com/fluxcd/flux2/discussions/4286#discussioncomment-7124267
